lib/rules/require-valid-css-selector-in-test-helpers.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const { getPropertyValue, parseCallee } = require('../utils/utils');
+const cssTree = require('css-tree');
+
+const TEST_HELPER_IMPORTS = new Set(['fillIn', 'findAll', 'find', 'click', 'focus']);
+const QUERY_SELECTOR_METHODS = new Set(['querySelectorAll', 'querySelector']);
+const PARENT_NODE_NAMES = new Set(['element', 'document']);
+const SELECTOR_RULES = Object.freeze({
+  unclosedAttr: {
+    hasError: (selector) => selector.includes('[') && !selector.includes(']'),
+    fix(node, selector, fixer) {
+      return fixer.replaceText(node.arguments[0], `'${selector}]'`);
+    },
+    errorMessage:
+      'Syntax error, you used an unclosed attribute selector: "{{selector}}", should be: "{{selector}}]"',
+  },
+  idStartsWithNumber: {
+    hasError: (selector) => selector.match(/^#\d/),
+    fix() {},
+    errorMessage: 'Syntax error, ids cannot start with a number: "{{selector}}"',
+  },
+  other: {
+    hasError: (selector) => !_isValidSelector(selector),
+    fix() {},
+    errorMessage: 'Syntax error, "{{selector}}" is not a valid selector',
+  },
+});
+
+function _isValidSelector(selector) {
+  try {
+    cssTree.parse(selector, {
+      context: 'selector',
+    });
+  } catch {
+    return false;
+  }
+
+  return true;
+}
+
+function _isAssertDomCall(node, assertIdentifierName) {
+  const calleeName = parseCallee(node) || [];
+
+  if (calleeName[1] !== 'dom' || !assertIdentifierName) {
+    return false;
+  }
+
+  return calleeName[0] === assertIdentifierName;
+}
+
+function _isQuerySelectorCall(node, isTestCall) {
+  const calleeName = parseCallee(node);
+
+  // only validate querySelector calls in the test context
+  if (!isTestCall) {
+    return false;
+  }
+
+  return QUERY_SELECTOR_METHODS.has(calleeName[1]) && PARENT_NODE_NAMES.has(calleeName[0]);
+}
+
+function _isTestHelperCall(node, hasTestHelperImport, localImportNames) {
+  const calleeName = parseCallee(node) || [];
+
+  return hasTestHelperImport && localImportNames.includes(calleeName[0]);
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow using invalid CSS selectors in test helpers',
+      category: 'Testing',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/require-valid-css-selector-in-test-helpers.md',
+    },
+    fixable: 'code',
+    schema: [],
+    messages: Object.keys(SELECTOR_RULES).reduce((accumulator, currentVal) => {
+      const _accumulator = accumulator || {};
+      _accumulator[currentVal] = SELECTOR_RULES[currentVal].errorMessage;
+
+      return _accumulator;
+    }, null),
+  },
+  create(context) {
+    let hasTestHelperImport = false;
+    let localImportNames = [];
+    let testImportLocalName = '';
+    let isTestCall = false;
+    let assertIdentifierName = '';
+
```

Comment:
Can you limit this rule to test files? And then add a test to ensure it doesn't run on non-test files.

```js
if (emberUtils.isTestFile(context.getFilename())) {
  // This rule does not apply to test files.
  return {};
}
```

Reply:
Done